apache/kyuubi
# Open modules to enable JDK 17 support

### _Why are the changes needed?_


After previous efforts, we are close to the goal. This is the latest step to to allow Kyuubi run on JDK 17 by adding `--add-open` in bootstrap scripts and test the plugin's parameter.

This PR also enables JDK 17 test in CI, but only for Kyuubi server and Spark(3.3+) engine moudles, because

- Spark supports JDK 17 since 3.3, see SPARK-33772
- Flink does not support JDK 17 until the latest 1.17

### _How was this patch tested?_


  - Run Kyuubi server on JDK 17
  - Run simple Spark (3.3.2) query on JDK 17
  - Run `kyuubi-zk-cli` on JDK 17

- [x] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request
